assets/js/offline-search.js

```diff
@@ -75,49 +75,70 @@
       maxPatternLength: 32,
       minMatchCharLength: 2,
       keys: [
-        "p",
-        "m",
-        "v",
-      ],
+        "n",    // type name
+        "p",    // property
+        "m",    // method
+        "v",    // enum (value)
+      ]
     };
 
     var fuse = new Fuse(data, options);
-    var results = fuse.search(query, { limit: 30 });
+    var results = fuse.search(query, { limit: 1000 });
 
     let resultCount = 0;
 
     for (const result of results) {
-      for (const match of result.matches) {
-        if (match.value.toLowerCase().indexOf(query) === -1) {
-          continue;
+      const $resultBaseRef = $('<a>').attr('href', '/' + result.item.id).text(result.item.n);
+      const $resultTitle = $('<strong>').append($resultBaseRef).append($('<span>').addClass('result-ctx').addClass(result.item.c));
+      const $resultList = $('<ul>');
+
+      const onlyTypeMatched = result.matches.length === 1 && result.matches[0].key === 'n';
+
+      if (!onlyTypeMatched) {
+        let matchCount = 0;
+        
+        for (const match of result.matches) {
+          if (match.value.toLowerCase().indexOf(query) === -1) {
+            continue;
+          }
+
+          if (match.key === 'n') {
+            continue;
+          }
+
+          let matchUrl = '/' + result.item.id;
+
+          if (++matchCount > 5) {
```

Comment:
We don't want to be hiding results. Even if the list is long all relevant results should be shown.